### ISSUE TITLE
LC_NUMERIC "C" and decimal conversion

### ISSUE THREAD

**Issue by rafagf1** [ruby-gnome2/ruby-gnome2#928]
I do not know if it's a bug or feature.

The "locale" with gtk3 2.2.5 are initialized to system locale, in my case "es_ES.UTF-8", and then makes a "setlocale( 'LC_NUMERIC', 'C')". In gtk3 3.0.9 this is not so and LC_NUMERIC maintains the value of the system locale, that in my case es_ES.UTF-8.

The problem is that there are some gems (such as "mysql2") using the "strtod()" function, and this function depends on the value of LC_NUMERIC to see which is the field decimals separator, that with "es_ES .UTF-8 " is the comma (,) instead of the dot (.)

mysql2 for example working properly before initializing gtk3, but after initialized gtk3 and try to pass the string  '1.25' to double with strtod() returns us '1.00' instead of 1.25

Gtk3 have removed setlocale ( 'LC_NUMERIC', 'C')  for some reason? 
Do these gems (such as mysql2) should use the "strtod_l" function instead of "strtod"?

``` Ruby
require 'ffi'
gem 'gtk3','=2.2.5'
#gem 'gtk3','=3.0.9'
require 'gtk3'


module Local
  extend FFI::Library
  ffi_lib FFI::Library::LIBC
  attach_function :setlocale, [:int, :string], :string
end


defs = %w[LC_CTYPE LC_NUMERIC LC_TIME LC_COLLATE LC_MONETARY LC_MESSAGES LC_ALL LC_PAPER LC_NAME LC_ADDRESS LC_TELEPHONE LC_MEASUREMENT LC_IDENTIFICATION]

puts "\nbefore GTK3\n"
defs.each_with_index { |d,i| puts "%s: %s\n" % [d,Local.setlocale( i, nil)] }
Gtk::Label.new
puts "\nafter GTK3\n"
defs.each_with_index { |d,i| puts "%s: %s\n" % [d,Local.setlocale( i, nil)] }
```

Thanks in advance

**Comment by kou** [comment]
Ah, we didn't notice that. Thanks.
The old behavior should be reimplemented.

**Comment by kou** [comment]
It's implemented by #947.
